### PR TITLE
Maintain filters on dashboard reload

### PR DESCRIPTION
**Pull request by julienreliefapps**
- Filter management improved

- Filter configuration of every pages saved.

- Now able to send a list of categories

### CODE REVIEW

File: src/js/actions/index.js [hunk 1]
```diff
@@ -153,21 +153,47 @@ export function changeCurrentLocale(locale) {
 
 // New Dashboard
 
+function fillParameterList(params = '', locationId = '') {
```
Comment (awalkowiak) [inline src/js/actions/index.js:156]: Name it `getParameterList`

File: src/js/components/tablero/Filter.jsx [hunk 2]
```diff
@@ -63,76 +75,101 @@ class Filter extends Component {
     this.setState({ filterCategorySelected: !this.state.filterCategorySelected });
   }
 
+  loadStoredFilters = (pageFilters) => {
+    if (this.props.pageFilters.length > 0) {
```
Comment (awalkowiak) [inline src/js/components/tablero/Filter.jsx:79]: Use destructuring at the begining.
`const { pageFilters, activeConfig } = this.props;`

File: src/js/components/tablero/Filter.jsx [hunk 3]
```diff
@@ -63,76 +75,101 @@ class Filter extends Component {
...
+      const listFilterSelected = [];
+
+      // Initialization of the page config
+      let pageConfig = JSON.parse(sessionStorage.getItem('pageConfig'));
+      if (!pageConfig) {
+        pageConfig = {};
+        sessionStorage.setItem('pageConfig', JSON.stringify(pageConfig));
+      }
+
+      if (!pageConfig[`${this.props.activeConfig}`]) {
```
Comment (awalkowiak) [inline src/js/components/tablero/Filter.jsx:89]: Destructure `activeConfig` from props at the beginning and use `pageConfig[activeConfig]`.

File: src/js/components/tablero/Filter.jsx [hunk 4]
```diff
@@ -63,76 +75,101 @@ class Filter extends Component {
...
+        pageConfig[`${this.props.activeConfig}`] = {};
```
Comment (awalkowiak) [inline src/js/components/tablero/Filter.jsx:90]: Same as above

File: src/js/components/tablero/Filter.jsx [hunk 5]
```diff
@@ -63,76 +75,101 @@ class Filter extends Component {
...
+      }
+      sessionStorage.setItem('pageConfig', JSON.stringify(pageConfig));
+
+      pageFilters.forEach((category) => {
+        const listFilterStored = JSON.parse(sessionStorage.getItem('pageConfig'))[`${this.props.activeConfig}`][`${category.name}`] || [];
```
Comment (awalkowiak) [inline src/js/components/tablero/Filter.jsx:95]: Same as above plus `[activeConfig][category.name]`

File: src/js/components/tablero/Filter.jsx [hunk 6]
```diff
@@ -63,76 +75,101 @@ class Filter extends Component {
...
+        listFilterStored.forEach((filter) => {
+          listFilterSelected.push(filter);
+        });
+      });
+      this.setState({ listFilterSelected });
+      this.setState({ filterAvailable: true });
```
Comment (awalkowiak) [inline src/js/components/tablero/Filter.jsx:101]: Combine these two setStates into one.

File: src/js/components/tablero/Filter.jsx [hunk 7]
```diff
@@ -63,76 +75,101 @@ class Filter extends Component {
...
-    listFilterToSend.push(valueCategory.id);
-    sessionStorage.setItem(nameCategory, JSON.stringify(listFilterToSend));
-    if (!sessionStorage.getItem('currentCategory')) {
-      sessionStorage.setItem('currentCategory', nameCategory);
+    if (!sessionStorage.getItem('pageConfig')) {
+      sessionStorage.setItem('pageConfig', JSON.stringify({}));
+    }
+
+    const listFilterToSend = JSON.parse(sessionStorage.getItem('pageConfig'));
+    if (!listFilterToSend[`${this.props.activeConfig}`][`${nameCategory}`]) {
```
Comment (awalkowiak) [inline src/js/components/tablero/Filter.jsx:119]: `listFilterToSend[activeConfig][nameCategory]`

File: src/js/components/tablero/Filter.jsx [hunk 8]
```diff
@@ -63,76 +75,101 @@ class Filter extends Component {
...
+      listFilterToSend[`${this.props.activeConfig}`][`${nameCategory}`] = [];
```
Comment (awalkowiak) [inline src/js/components/tablero/Filter.jsx:120]: `listFilterToSend[activeConfig][nameCategory]`

File: src/js/components/tablero/Filter.jsx [hunk 9]
```diff
@@ -63,76 +75,101 @@ class Filter extends Component {
...
     }
 
+    listFilterToSend[`${this.props.activeConfig}`][`${nameCategory}`].push({
```
Comment (awalkowiak) [inline src/js/components/tablero/Filter.jsx:123]: `listFilterToSend[activeConfig][nameCategory]`

File: src/js/components/tablero/Filter.jsx [hunk 10]
```diff
@@ -63,76 +75,101 @@ class Filter extends Component {
...
   removeFilterFromList = (key) => {
     const actualList = this.state.listFilterSelected;
     const elementToDelete = actualList[key];
 
     // Management of the filterList in the session storage
-    const actualFilterList = JSON.parse(sessionStorage.getItem(elementToDelete[0])) || [];
-    const newFilterList = actualFilterList.filter(item => item !== elementToDelete[1].id);
-    sessionStorage.setItem(elementToDelete[0], JSON.stringify(newFilterList));
+    const newFilterList = JSON.parse(sessionStorage.getItem('pageConfig'));
+    newFilterList[`${this.props.activeConfig}`][`${elementToDelete.nameCategory}`] = newFilterList[`${this.props.activeConfig}`][`${elementToDelete.nameCategory}`]
```
Comment (awalkowiak) [inline src/js/components/tablero/Filter.jsx:143]: `newFilterList[activeConfig][elementToDelete.nameCategory]`

File: src/js/components/tablero/Filter.jsx [hunk 11]
```diff
@@ -63,76 +75,101 @@ class Filter extends Component {
...
     // Management of the list of the filter in the DOM
     const newList = actualList.slice(0, key).concat(actualList.slice(key + 1, actualList.length));
     this.setState({ listFilterSelected: newList });
 
     // Removing current category if no filter selected
-    if (JSON.parse(sessionStorage.getItem(elementToDelete[0])).length === 0) {
-      sessionStorage.removeItem('currentCategory');
-      sessionStorage.removeItem(elementToDelete[0]);
+    const pageConfig = JSON.parse(sessionStorage.getItem('pageConfig'));
+    if (pageConfig[`${this.props.activeConfig}`][`${elementToDelete.nameCategory}`].length === 0) {
```
Comment (awalkowiak) [inline src/js/components/tablero/Filter.jsx:153]: `pageConfig[activeConfig][elementToDelete.nameCategory]`

File: src/js/components/tablero/Filter.jsx [hunk 12]
```diff
@@ -63,76 +75,101 @@ class Filter extends Component {
...
+      delete pageConfig[`${this.props.activeConfig}`][`${elementToDelete.nameCategory}`];
```
Comment (awalkowiak) [inline src/js/components/tablero/Filter.jsx:154]: `pageConfig[activeConfig][elementToDelete.nameCategory]`

**Review by awalkowiak — CHANGES_REQUESTED** [review]

File: src/js/actions/index.js [hunk 13]
```diff
@@ -153,21 +153,47 @@ export function changeCurrentLocale(locale) {
...
+  const listFiltersSelected = [];
+  const listValues = [];
+
+  const dashboardKey = sessionStorage.getItem('dashboardKey');
+
+  const pageConfig = JSON.parse(sessionStorage.getItem('pageConfig')) || {};
+
+  if (!pageConfig[`${dashboardKey}`]) { pageConfig[`${dashboardKey}`] = {}; }
```
[the remark this answers is not in the document]
Comment (julienreliefapps) [reply src/js/actions/index.js]: Done

File: src/js/components/tablero/Filter.jsx (hunk 2 above)
Comment (julienreliefapps) [reply src/js/components/tablero/Filter.jsx]: Done

File: src/js/components/tablero/Filter.jsx (hunk 3 above)
Comment (julienreliefapps) [reply src/js/components/tablero/Filter.jsx]: Done

File: src/js/components/tablero/Filter.jsx (hunk 8 above)
Comment (julienreliefapps) [reply src/js/components/tablero/Filter.jsx]: Done

File: src/js/components/tablero/Filter.jsx (hunk 9 above)
Comment (julienreliefapps) [reply src/js/components/tablero/Filter.jsx]: Done

File: src/js/components/tablero/Filter.jsx (hunk 10 above)
Comment (julienreliefapps) [reply src/js/components/tablero/Filter.jsx]: Done

File: src/js/components/tablero/Filter.jsx (hunk 11 above)
Comment (julienreliefapps) [reply src/js/components/tablero/Filter.jsx]: Done

File: src/js/components/tablero/Filter.jsx (hunk 12 above)
Comment (julienreliefapps) [reply src/js/components/tablero/Filter.jsx]: Done

File: src/js/components/tablero/Filter.jsx (hunk 6 above)
Comment (awalkowiak) [reply src/js/components/tablero/Filter.jsx:103]: @julienreliefapps how about this one?

**Review by awalkowiak — APPROVED** [review]

**Pull request merged**